tutorials/animation/playing_videos.rst

```diff
@@ -0,0 +1,226 @@
+as it was too buggy and difficult to maintain.
+
+.. note::
+
+    You may find videos with ``.ogg`` or ``.ogx`` extension, which are generic
+    extensions for data within an Ogg container.
+
+    Renaming this file extension to ``.ogv`` *may* allow the video to be
+    imported in Godot. However, not all files with ``.ogg`` or ``.ogx``
+    extensions are videos - some of them may only contain audio.
+
+Setting up VideoStreamPlayer
+----------------------------
+
+1. Create a VideoStreamPlayer node using the Create New Node dialog.
+2. Select the VideoStreamPlayer node in the scene tree dock, go to the inspector
+   and load an ``.ogv`` file in the Stream property.
+
+   - If you don't have your video in Ogg Theora format yet, jump to
+     :ref:`doc_playing_videos_recommended_theora_encoding_settings`.
+
+3. If you wish the video to play as soon as the scene is loaded, check
+   **Autoplay** in the inspector. If not, leave **Autoplay** disabled and call
+   ``play()`` on the VideoStreamPlayer node in a script to start playback when
+   desired.
+
+Handling resizing and different aspect ratios
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default in 4.0, the VideoStreamPlayer will automatically be resized to match
+the video's resolution. You can make it follow usual Control sizing by enabling
+**Expand** on the VideoStreamPlayer node.
+
+To adjust how the VideoStreamPlayer node resizes depending on window size,
+adjust the anchors using the **Layout** menu at the top of the 2D editor
+viewport. However, this setup may not be powerful enough to handle all use
+cases, such as playing fullscreen videos without distorting the video (but with
+empty space on the edges instead). Instead, you can use an
+:ref:`class_AspectRatioContainer` node which is designed to handle this kind of
+use case.
+
+Add an AspectRatioContainer node. Make sure it is not a child of any other
+container node. Select the AspectRatioContainer node, then set its **Layout** at
+the top of the 2D editor to **Full Rect**. Set **Ratio** in the
+AspectRatioContainer node to match your video's aspect ratio. You can use math
+formulas in the inspector to help yourself. Remember to make one of the operands
+a float. Otherwise, the division's result will always be an integer.
+
+.. figure:: img/playing_videos_aspect_ratio_container.png
+   :figclass: figure-w480
+   :align: center
+   :alt: AspectRatioContainer's Ratio property being modified in the editor inspector
+
+   This will evaluate to (approximately) 1.777778
+
+
+Once you've configured the AspectRatioContainer, reparent your VideoStreamPlayer
+node to be a child of the AspectRatioContainer node. Make sure **Expand** is
+disabled on the VideoStreamPlayer. Your video should now scale automatically
+to fit the whole screen while avoiding distortion.
+
+.. seealso::
+
+    See :ref:`doc_multiple_resolutions` for more tips on supporting multiple
+    aspect ratios in your project.
+
+Displaying a video on a 3D surface
```

Comment:
I kinda disagree in this specific case, as for a in-engine video player, you really have the two use cases of 1.) cut scenes 2.) play video in 3D e.g. for displays.

So explaining how to do both seems worthwhile if its possible in a succinct way, as it basically solves 99% of the problem space of this feature. Thats usually not the case with other features, i.e. doing a comprehensive tutorial for networking or particles or something is hard to impossible because there is so much you may want to do with those, so the scope if too large for the docs.

I agree it somewhat goes against the "manual, not tutorials" aim, but IMO not too much that its not worth it. I'm happy to get more input on this however :)

Reply:
Displaying a video on a 3D surface is a common question, so I'd prefer to have it documented in the manual (even if it borders on tutorial territory).

Reply:
I'll give approval for this PR, but in the future I'd like to revisit this if significant improvements are made to the viewport page.